GoogleDataTransport/GDTCORLibrary/GDTCORFlatFileStorage.m

```diff
@@ -72,6 +100,66 @@ + (NSString *)libraryDataPath {
   return libraryDataPath;
 }
 
++ (NSDictionary<NSString *, NSString *> *)pathsForEvent:(GDTCOREvent *)event {
+  NSString *dataPath = [NSString
+      stringWithFormat:@"%@/%@", [GDTCORFlatFileStorage baseEventStoragePath], event.eventID];
+  NSString *mappingIDPath =
+      [NSString stringWithFormat:@"%@/%@/%ld/%@", [GDTCORFlatFileStorage baseEventStoragePath],
+                                 event.mappingID, (long)event.qosTier, event.eventID];
+  NSString *qosTierPath = [NSString
+      stringWithFormat:@"%@/%ld/%ld/%@/%@", [GDTCORFlatFileStorage baseEventStoragePath],
+                       (long)event.target, (long)event.qosTier, event.mappingID, event.eventID];
+  return @{
+    gGDTCORFlatFileStorageEventDataPathKey : dataPath,
+    gGDTCORFlatFileStorageMappingIDPathKey : mappingIDPath,
+    gGDTCORFlatFileStorageQoSTierPathKey : qosTierPath
+  };
+}
+
++ (NSString *)pathForTarget:(GDTCORTarget)target
+                    qosTier:(nullable NSNumber *)qosTier
+                  mappingID:(nullable NSString *)mappingID {
+  NSString *baseEventPath = [GDTCORFlatFileStorage baseEventStoragePath];
+  // If only a target was given, return the target path.
+  if (qosTier == nil && mappingID == nil) {
+    return [NSString stringWithFormat:@"%@/%ld", baseEventPath, (long)target];
+  }
+
+  // If only a target and mappingID were given, return the mapping ID path.
+  if (qosTier == nil) {
+    return [NSString stringWithFormat:@"%@/%ld/%@", baseEventPath, (long)target, mappingID];
+  }
+
+  // If only a target and qosTier were given, return the QoS tier path.
+  if (mappingID == nil) {
+    return [NSString stringWithFormat:@"%@/%ld/%@", baseEventPath, (long)target, qosTier];
+  }
+
+  // If a target, mappingID, and qosTier were all given, return a single target/qosTier/mappingID
+  // directory.
+  return
+      [NSString stringWithFormat:@"%@/%ld/%@/%@", baseEventPath, (long)target, qosTier, mappingID];
+}
+
++ (nullable NSArray<NSString *> *)searchPathsWithEventSelector:
```

Comment:
Is `nullable` actually required here? Doesn't look like the method ever returns `nil`

Reply:
Removed nullability.